src/core/nixl_etcd_metadata_backend.cpp

```
@@ -0,0 +1,382 @@
+        catch (const std::exception &e) {
+            NIXL_ERROR << "Error fetching key: " << metadata_key << " from etcd: " << e.what();
+            return NIXL_ERR_UNKNOWN;
+        }
+    }
+
+    nixl_status_t
+    waitForMetadataFromEtcd(const std::string &metadata_key, nixl_blob_t &remote_metadata) {
+        try {
+            etcd::Response response = etcd->get(metadata_key);
+            int64_t watch_index = response.index();
+            std::promise<nixl_status_t> ret_prom;
```

Cppcheck (2.21.0)

[style] The function 'addErrorTelemetry' is never used.

(unusedFunction)